### PR TITLE
runtime(cgdbrc): add support

### PR DESCRIPTION
**Pull request by Freed-Wu**

### CODE REVIEW

File: runtime/syntax/cgdbrc.vim [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+" Documentation:        https://cgdb.github.io/docs/Configuring-CGDB.html
+" Latest Revision:      2024-04-09
+
+if exists('b:current_syntax')
+  finish
+endif
+let b:current_syntax = 'cgdbrc'
+
+" Read the Vim syntax to start with
+if version < 600
```
Comment (chrisbra) [inline runtime/syntax/cgdbrc.vim]: I think we do not need the version test for vim < 6.0 :)

Comment (Freed-Wu) [reply runtime/syntax/cgdbrc.vim]: Fixed. It is copied from https://github.com/Freed-Wu/vim/blob/master/runtime/syntax/arduino.vim#L25

File: runtime/ftplugin/cgdbrc.vim [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  finish
+endif
+let b:did_ftplugin = 1
+
+let s:save_cpoptions = &cpoptions
+set cpoptions&vim
+
+let b:undo_ftplugin = 'setl <cms'
+
+setlocal commentstring=#%s
```
Comment (dkearns) [inline runtime/ftplugin/cgdbrc.vim:17]: Can you please set `'comments'` as well?

**Review by dkearns — CHANGES_REQUESTED** [review]

**Comment by chrisbra** [comment]
thanks!

**Pull request closed without merging**